toolsrc/src/vcpkg-test/system.cpp

```diff
@@ -48,7 +55,10 @@ namespace
             tmp.append(unpacked);
         }
 
-        const int exit_code = putenv(tmp.c_str());
+        // putenv expects the string to never go out of scope
+        char* env_string = new char[tmp.size() + 1];
+        strcpy(env_string, tmp.c_str());
```

Comment:
```suggestion
        memcpy(env_string, tmp.c_str(), tmp.size() + 1);
```

(due to `strcpy` being a banned API)